controllers/multiclusterhub_controller.go

```diff
@@ -146,7 +147,42 @@ func (r *MultiClusterHubReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			retError = statusError
 		}
 	}()
+	matchCurrent, _ := regexp.MatchString("^2.4", multiClusterHub.Status.CurrentVersion)
+	matchDesired, _ := regexp.MatchString("^2.5", multiClusterHub.Status.DesiredVersion)
 
+	if (matchCurrent) && (matchDesired) {
```

Comment:
This is getting called too soon in the reconcile loop. It should not come before the finalizer logic, otherwise it could potentially block uninstalls. I would move it after the pause logic so we don't try modifying things when the mch is paused.